### PR TITLE
feat(jupiter): implement api logging support for jupiter

### PR DESCRIPTION
**Pull request by jhaeyaert**
Implements the logging feature for Jupiter.

image

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/jupiter-logging/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-gxxugmggty.chromatic.com)

### CODE REVIEW

File: gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LoggingContext.java [hunk 1]
```diff
@@ -24,21 +25,28 @@
...
+    private final Logging logging;
     private int maxSizeLogMessage = -1;
     private String excludedResponseTypes;
 
     public LoggingContext(final Logging logging) {
         this.logging = logging;
     }
 
+    @Override
+    public String getCondition() {
```
Comment (guillaumelamirand) [inline gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LoggingContext.java:46]: Does that mean we can now apply a condition on top of a logger ?

**Review by guillaumelamirand — CHANGES_REQUESTED** [review]

File: gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LoggingContext.java (hunk 1 above)
Comment (jhaeyaert) [reply gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LoggingContext.java:46]: Defining a condition on the logging is something already existing in v3 and has just been reimplemented with jupiter.
image

Or maybe I didn't understood your question 🧐 ?

Comment (jhaeyaert) [reply gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LoggingContext.java:46]: To provide more details regarding the add `getCondition()`, I reuse the `ExpressionLanguageConditionFilter` which just requires a `ConditionSupplier` to access the condition. For that, I just had to expose the already existing condition located in the `LoggingContext` to make things working pretty smoothly 🙂

Comment (guillaumelamirand) [reply gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LoggingContext.java:46]: I wasn't aware of this current capability, thank for the answer

**Review by guillaumelamirand — APPROVED** [review]

**Pull request merged**